src/Raven.Server/Documents/Sharding/ShardedDocumentPutAction.cs

```diff
@@ -37,3 +54,21 @@ protected override unsafe void WriteSuffixForIdentityPartsSeparator(ref char* va
             valueWritePosition[j + 1] = idSuffixPtr[j];
     }
 }
+
+public class BucketUnderMoveException : ConcurrencyException
```

Comment:
move that to exceptions in server namespace

Reply:
removed, currently not used